ios/AnalyticsModule.swift

```diff
@@ -0,0 +1,178 @@
+     */
+    @objc(attach:playerId:)
+    func attach(_ nativeId: NativeId, playerId: NativeId) {
+        bridge.uiManager.addUIBlock { [weak self] _, _ in
+            guard
+                let collector = self?.collectors[nativeId],
+                let player = self?.playerModule?.retrieve(playerId)
+            else {
+                return
+            }
+            collector.attachPlayer(player: player)
+        }
+    }
+
+    /**
+     Detaches the player object from a `BitmovinPlayerCollector` instance.
+     - Parameter nativeId: Native Id of the collector instance.
+     */
+    @objc(detach:)
+    func detach(_ nativeId: NativeId) {
+        bridge.uiManager.addUIBlock { [weak self] _, _ in
+            guard let collector = self?.collectors[nativeId] else {
+                return
+            }
+            collector.detachPlayer()
+        }
+    }
+
+    /**
+     Updates the custom data config for a `BitmovinPlayerCollector` instance.
+     - Parameter nativeId: Native Id of the collector instance.
+     - Parameter json: Custom data config json.
+     */
+    @objc(setCustomDataOnce:json:)
+    func setCustomDataOnce(_ nativeId: NativeId, json: Any?) {
+        bridge.uiManager.addUIBlock { [weak self] _, _ in
+            guard
+                let collector = self?.collectors[nativeId],
+                let customData = RCTConvert.analyticsCustomData(json)
+            else {
+                return
+            }
+            collector.setCustomDataOnce(customData: customData)
+        }
+    }
+
+    /**
+     Sets the custom data config for a `BitmovinPlayerCollector` instance.
+     - Parameter nativeId: Native Id of the collector instance.
+     - Parameter json: Custom data config json.
+     */
+    @objc(setCustomData:json:)
+    func setCustomData(_ nativeId: NativeId, json: Any?) {
+        bridge.uiManager.addUIBlock { [weak self] _, _ in
+            guard
+                let collector = self?.collectors[nativeId],
+                let customData = RCTConvert.analyticsCustomData(json)
+            else {
+                return
+            }
+            collector.setCustomData(customData: customData)
+        }
+    }
+
+    /**
+     Gets the current custom data config for a `BitmovinPlayerCollector` instance.
+     - Parameter nativeId: Native Id of the the collector instance.
+     - Parameter resolver: JS promise resolver.
+     - Parameter rejecter: JS promise rejecter.
+     */
+    @objc(getCustomData:resolver:rejecter:)
+    func getCustomData(
+        _ nativeId: NativeId,
+        resolver resolve: @escaping RCTPromiseResolveBlock,
+        rejecter reject: @escaping RCTPromiseRejectBlock
+    ) {
+        bridge.uiManager.addUIBlock { [weak self] _, _ in
+            guard
+                let collector = self?.collectors[nativeId],
+                let customData = RCTConvert.toJson(analyticsCustomData: collector.getCustomData())
+            else {
+                return
```

Comment:
Should we call `reject` here to not keep the promise unresolved in case collector is not found?

Reply:
76321d08a16288322f832c2a23b0d6519de671ae